File: src/codegen/java/templates/RequestDescription.java
import java.util.Collections;
import java.util.List;
import java.util.Map;

public interface RequestDescription<Body> {
  default String method() {
    return "GET";
  }

  String path();

  default Map<String, String> pathParams() {
    return Collections.emptyMap();
  }

  void query(String name, String value);
  default Map<String, List<String>> query() {
    return Collections.emptyMap();
  }

  void header(String name, String value);
  default Map<String, List<String>> headers() {
    return Collections.emptyMap();
  }

  default Body body() {
    return null;
  }
}

File: src/codegen/php/templates/RequestDescription.php
interface RequestDescription {
    public function method(): string;
    public function path(): string;
    public function pathParams(): array;

    public function setQuery(string $name, string $value);

    public function query(): array;
    public function setHeader(string $name, string $value);
    public function headers(): array;
    public function body(): mixed;
}